client/changelist/changelist.go

```diff
@@ -32,3 +36,30 @@ func (cl *memChangelist) Clear(archive string) error {
 func (cl *memChangelist) Close() error {
 	return nil
 }
+
+func (cl *memChangelist) NewIterator() (ChangeIterator, error) {
+	return &MemChangeListIterator{index: 0, collection: cl.changes}, nil
```

Comment:
The `index: 0` can be omited because fields default to the 0 value, but it doesn't hurt to have it here.